psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -235,6 +235,110 @@ void Wavefunction::deep_copy(const Wavefunction *other)
 
 }
 
+void Wavefunction::c1_deep_copy(SharedWavefunction other)
+{
+    c1_deep_copy(other.get());
+}
+
+void Wavefunction::c1_deep_copy(const Wavefunction *other)
```

Comment:
It might be better if this simply returned a new Wavefunction (make this a static method).